pyronear_ds/datasets/utils.py

```diff
@@ -1,18 +1,198 @@
+    elif len(list_extensions) == 1:
+        extension = list_extensions[0]
+    else:
+        raise ValueError(f'Error {url} contains more than one extension')
+
+    if len(list_compressions) == 0:
+        compression = None
+
+    elif len(list_compressions) == 1:
+        compression = list_compressions[0]
+
+    elif len(list_compressions) == 2:
+        compression = "tar.gz"
+
+    else:
+        raise ValueError(f'Error {url} contains more than one compression format')
+
+    return (base, extension, compression)
+
+
+def download(url: str, default_extension: str, unzip: Optional[bool] = True, destination: Optional[str] = './tmp'):
+    """Helper function for downloading, unzipping and saving compressed file from a given URL.
+
+    Args:
+        url: URL of the compressed archive
+        default_extension: extension of the archive
+        unzip: whether archive should be unzipped. Defaults to True.
+        destination: folder where the file should be saved. Defaults to '.'.
+    """
+# TODO Write case tests for zip, tar.gz, gz and uncompressed files
+# Check if the destination directory is created each if not exist
+# Check if the file are  download
+# Add print and logging statement add
+    base, extension, compression = get_fname(url)
+
+    if unzip and compression == "zip":
+        content = url_retrieve(url)
+        os.makedirs(os.path.dirname(destination), exist_ok=True)
+        with ZipFile(BytesIO(content)) as zip_file:
+            zip_file.extractall(destination)
+
+    elif unzip and compression == "tar.gz":
+        content = url_retrieve(url)
+        os.makedirs(os.path.dirname(destination), exist_ok=True)
+        with tarfile.open(fileobj=BytesIO(content), mode='r:gz') as tar_file:
+            tar_file.extractall(path=destination)
+
+    elif unzip and compression == "gz":
+        content = url_retrieve(url)
+        file_name = f"{base}.{extension}" if extension is not None else f"{base}.{default_extension}"
+        full_path = os.path.join(destination, file_name)
+        os.makedirs(os.path.dirname(full_path), exist_ok=True)
+        with gzip.open(BytesIO(content)) as gzip_file, open(full_path, 'wb+') as unzipped_file:
+            shutil.copyfileobj(gzip_file, unzipped_file)
+
+    elif not unzip and compression is None:
+        content = url_retrieve(url)
+        file_name = f"{base}.{extension}" if extension is not None else f"{base}.{default_extension}"
+        full_path = os.path.join(destination, file_name)
+        os.makedirs(os.path.dirname(full_path), exist_ok=True)
+        with open(full_path, 'wb+') as file:
+            file.write(content)
+
+    elif not unzip and isinstance(compression, str):
+        content = url_retrieve(url)
+        file_name = f"{base}.{compression}"
+        full_path = os.path.join(destination, file_name)
+        os.makedirs(os.path.dirname(full_path), exist_ok=True)
+        with open(full_path, 'wb+') as file:
+            file.write(content)
```

Comment:
Perhaps some refactoring could be useful here?

Reply:
Yes !

Reply:
Just this one left and we're good to go I think!

Reply:
For this one I don't see any clear-cut way of refactoring it.

- I could write a specific function and test just for the case 3 and 4 but it doesn't seem to help 
- Or I could write a more general function but this would amount to moving the elif blocks to another function

Anyway, I left the function as is

Reply:
You're right, I think we could still have just one shared command :
moving this before the conditional statements
```
content = url_retrieve(url)
```